manifests/repository/debian.pp

```diff
@@ -30,5 +30,9 @@
 
   include apt
 
-  Apt::Source['opensearch'] ~> Class['apt::update']
+  if $opensearch::manage_package {
+    Apt::Source['opensearch'] ~> Class['apt::update'] -> Package['opensearch']
```

Comment:
https://github.com/puppetlabs/puppetlabs-apt#update-the-list-of-packages shouldn't you notify the exec resource, not the class?

Reply:
Done.